UIKit/S2MFoldAnimator.m

```diff
@@ -0,0 +1,417 @@
+//
+//  S2MFoldAnimator.m
+//  S2MToolbox
+//
+//  Created by François Benaiteau on 23/07/14.
+//  Copyright (c) 2014 SinnerSchrader Mobile. All rights reserved.
+//
+
+
+#import "S2MFoldAnimator.h"
+
+@implementation S2MFoldAnimator
+
+- (id)init
+{
+    self = [super init];
+    if (self) {
+        self.folds = 1;
+        self.unfolding = YES;
+        self.direction = S2MFoldAnimatorDirectionRightToLeft;
+    }
+    return self;
+}
+
+#pragma mark - Public
+
+
+- (void)animateWithDuration:(NSTimeInterval)duration
+                       view:(UIView *)view
+              containerView:(UIView*)containerView
+                 completion:(void (^)(BOOL finished))completion
+{
+    __block CGPoint viewOrigin;
+    __block CGSize viewSize;
+    __block CGFloat viewFoldMeasure;
+    __block NSMutableArray* viewFolds;
+    [UIView performWithoutAnimation:^{
+        if (view.superview == nil) {
+            [containerView addSubview:view];
+        }
+        viewSize = view.bounds.size;
+        viewOrigin = view.frame.origin;
+        // Add a perspective transform
+        CATransform3D transform = CATransform3DIdentity;
+        transform.m34 = -0.005;
+        containerView.layer.sublayerTransform = transform;
```

Comment:
@joernito more infos here : https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/CoreAnimation_guide/AdvancedAnimationTricks/AdvancedAnimationTricks.html#//apple_ref/doc/uid/TP40004514-CH8-SW13

Actually we could make it as a property as one might want to adjust this